app/src/main/java/app/revanced/tiktok/feedfilter/FeedItemsFilter.java

```diff
@@ -11,28 +12,31 @@
 public class FeedItemsFilter {
 
     public static void filter(FeedItemList feedItemList) {
-        if (SettingsEnum.TIK_REMOVE_ADS.getBoolean()) removeAds(feedItemList);
-        if (SettingsEnum.TIK_HIDE_LIVE.getBoolean()) removeLive(feedItemList);
-    }
-
-    private static void removeAds(FeedItemList feedItemList) {
-        List<Aweme> items = feedItemList.items;
-        Iterator<Aweme> it = items.iterator();
-        while (it.hasNext()) {
-            Aweme item = it.next();
-            if (item != null && (item.isAd() || item.isWithPromotionalMusic())) {
-                it.remove();
-            }
-        }
-    }
+        boolean ads = SettingsEnum.TIK_REMOVE_ADS.getBoolean();
+        boolean live = SettingsEnum.TIK_HIDE_LIVE.getBoolean();
+        boolean story = SettingsEnum.TIK_HIDE_STORY.getBoolean();
+        boolean image = SettingsEnum.TIK_HIDE_IMAGE.getBoolean();
+        long minView = Long.parseLong(SettingsEnum.TIK_HIDE_PLAY_COUNT.getString().split("-")[0]);
+        long maxView = Long.parseLong(SettingsEnum.TIK_HIDE_PLAY_COUNT.getString().split("-")[1]);
+        long minLike = Long.parseLong(SettingsEnum.TIK_HIDE_LIKE_COUNT.getString().split("-")[0]);
+        long maxLike = Long.parseLong(SettingsEnum.TIK_HIDE_LIKE_COUNT.getString().split("-")[1]);
 
-    private static void removeLive(FeedItemList feedItemList) {
         List<Aweme> items = feedItemList.items;
         Iterator<Aweme> it = items.iterator();
         while (it.hasNext()) {
             Aweme item = it.next();
-            if (item != null && (item.isLive() || item.isLiveReplay())) {
-                it.remove();
+            if (item != null) {
+                if (ads && (item.isAd() || item.isWithPromotionalMusic())) it.remove();
```

Comment:
Instead of doing multiple ifs you should create an abstract class that has a check method that takes the item as a parameter. The abstract class is initialized with the respective setting and the implementation to check for isLive, getIsTikTokStory and so on respectively.

Reply:
Once that is done, you can then instantiate a list of this abstract class implementations and then check if any of these return true for the check method and filter

Reply:
I have pushed the changes. Please tell me if there are any other things that need improvement.

Reply:
I'll add some last refactoring changes then we should be gtg once other reviews are resolved.